src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -463,5 +454,25 @@ private static bool RequiredDelimiterIsMissing(SyntaxNode currentNode)
                 currentNode.GetParentheses().closeParen.IsMissing ||
                 currentNode.GetBraces().closeBrace.IsMissing;
         }
+
+        private static bool HasDelimitersButCaretIsOutside(SyntaxNode currentNode, int caretPosition)
+        {
+            if (currentNode.GetParentheses() is ((not SyntaxKind.None) openParenthesis, (not SyntaxKind.None) closeParenthesis))
```

Comment:
📝 This syntax allows us to use the `Deconstruct` extension for `SyntaxToken` and check directly against `SyntaxKind`. However, I wasn't able to find a syntax which allowed us to _both_ deconstruct the tuple elements _and_ reuse the target variable. For example, I could not do this:

```cs
private static bool HasDelimitersButCaretIsOutside(SyntaxNode currentNode, int caretPosition)
{
    if (currentNode.GetParentheses() is ((not SyntaxKind.None) openDelimiter, (not SyntaxKind.None) closeDelimiter)
        || currentNode.GetBrackets() is ((not SyntaxKind.None) openDelimiter, (not SyntaxKind.None) closeDelimiter)
        || currentNode.GetBraces() is ((not SyntaxKind.None) openDelimiter, (not SyntaxKind.None) closeDelimiter))
    {
        return openDelimiter.SpanStart >= caretPosition || closeDelimiter.Span.End <= caretPosition;
    }
    else
    {
        return false;
    }
}
```

Reply:
i don't love it, but i can't think of anything better :)